File: PHM_claude/phm_pipeline/domain/shared/boundary.py
"""Lightweight API/domain markers.

These markers are intentionally behavior-neutral. Phase 1 keeps the same
database and runtime, but makes Edge / Cloud / Shared ownership explicit.
"""
from __future__ import annotations

from enum import Enum
from typing import Callable


class Domain(str, Enum):
    EDGE = "Edge"
    CLOUD = "Cloud"
    SHARED = "Shared"


def tag_api(domain: Domain, purpose: str = "") -> Callable:
    """Attach domain metadata to a Flask view without changing behavior."""
    def deco(fn: Callable) -> Callable:
        setattr(fn, "api_domain", domain.value)
        setattr(fn, "api_purpose", purpose)
        return fn

    return deco

File: 数控机床数据采集与状态监测系统/WebDashboard/api/src/domain/boundary.js
// Phase 1 domain markers. These helpers attach metadata to Express handlers
// without changing request behavior.
export const ApiDomain = Object.freeze({
  EDGE: 'Edge',
  CLOUD: 'Cloud',
  SHARED: 'Shared',
});

export function markApi(domain, purpose, handler) {
  handler.apiDomain = domain;
  handler.apiPurpose = purpose || '';
  return handler;
}

export const edgeApi = (purpose, handler) => markApi(ApiDomain.EDGE, purpose, handler);
export const cloudApi = (purpose, handler) => markApi(ApiDomain.CLOUD, purpose, handler);
export const sharedApi = (purpose, handler) => markApi(ApiDomain.SHARED, purpose, handler);

